### PR TITLE
camera: Updates for the imminent updated Bullseye release

### PR DESCRIPTION
**Pull request by davidplowman**

### CODE REVIEW

**Comment by davidplowman** [comment]
@naushir

**Comment by naushir** [comment]
LGTM

File: documentation/asciidoc/accessories/camera/camera_usage.adoc [hunk 1]
```diff
@@ -34,6 +34,17 @@ Reasons to consider staying with an older OS and using the legacy _Raspicam_ sta
 
 Apart from some general information on setting up the camera, most of this documentation is divided into sections on xref:camera.adoc#libcamera-and-libcamera-apps[libcamera and the libcamera-apps], and the xref:camera.adoc#raspicam-commands[legacy _Raspicam_ apps]. Once you've xref:camera.adoc#installing-a-raspberry-pi-camera[installed your camera module], please click on one of these two links.
 
+==== Re-enabling the legacy stack
+
+The legacy camera stack can be re-enabled in Bullseye using the following steps.
+
+1. Ensure you system is up-to-date and reboot it.
```
Comment (lurch) [inline documentation/asciidoc/accessories/camera/camera_usage.adoc:41]: "your system"?

File: documentation/asciidoc/accessories/camera/libcamera_options_common.adoc [hunk 2]
```diff
@@ -19,6 +19,23 @@ libcamera-apps build: ca559f46a97a 27-09-2021 (14:10:24)
 libcamera build: v0.0.0+3058-c29143f7
 ----
 
+----
+	--config,	-c		Read options from the given file <filename>
+----
+
+Normally options are read from the command line, but in case multiple options are required it may be more convenient to keep them in a file.
+
+Example: `libcamera-hello -c config.txt`
```
Comment (lurch) [inline documentation/asciidoc/accessories/camera/libcamera_options_common.adoc:28]: I wonder if `config.txt` is a bad-choice here, given the filename-overlap with https://www.raspberrypi.com/documentation/computers/config_txt.html :wink: 
Perhaps `camera-options.txt` or something?

File: documentation/asciidoc/accessories/camera/libcamera_options_common.adoc [hunk 3]
```diff
@@ -19,6 +19,23 @@ libcamera-apps build: ca559f46a97a 27-09-2021 (14:10:24)
...
+
+This is a text file containing individual lines of `key=value` pairs, for example:
+
+----
+timeout=99000
+verbose=
+----
+
+Note how the `=` is required even for implicit options, and that the `--` used on the command line are omitted.
```
Comment (lurch) [inline documentation/asciidoc/accessories/camera/libcamera_options_common.adoc:37]: Does it support both long- and short-options, or is it long-options only?

File: documentation/asciidoc/accessories/camera/libcamera_options_common.adoc [hunk 4]
```diff
@@ -145,6 +162,25 @@ Using this option often incurs a frame rate penalty, as larger resolution frames
...
+----
+	--mode				Specify sensor mode, given as <width>:<height>:<bit-depth>:<packing>
+----
+
+This option is more general than `--rawfull` and allows the precise selection of one of the camera modes. The mode should be specified by giving its width, height, bit-depth and packing, separated by colons. These numbers do not have to be exact as the system will select the closest it can find. Moreover, the bit-depth and packing are optional (defaulting to 12 and `P` for "packed" respectively). For example:
+
+* `4056:3040:12:P` - 4056x3040 resolution, 12 bits per pixel, packed. This means that raw image buffers will be packed so that 2 pixel values occupy only 3 bytes.
+* `1632:1224:10` - 1632x1224 resolution, 10 bits per pixel. It will default to "packed". A 10 bit packed mode would store 4 pixels in every 5 bytes.
+* `2592:1944:10:U` - 2592x1944 resolution, 10 bits per pixel, unpacked. An unpacked format will store every pixel in 2 bytes, in this case with the top 6 bits of each value being zero.
+* `3264:2448` - 3264x2448 resolution. It will try to select the default 12 bit per mode but in the case of the v2 camera there isn't one, so a 10 bit mode would be chosen instead.
```
Comment (lurch) [inline documentation/asciidoc/accessories/camera/libcamera_options_common.adoc:174]: "12 bit per mode" -> "12 bit mode"?  Also, should "12 bit" be hyphenated as 12-bit ?

File: documentation/asciidoc/accessories/camera/libcamera_options_common.adoc [hunk 5]
```diff
@@ -145,6 +162,25 @@ Using this option often incurs a frame rate penalty, as larger resolution frames
...
+
+The `--mode` option affects the mode choice for video recording and still capture. To control the mode choice during the preview phase prior to still capture, please use the `--viewfinder-mode` option.
+
+----
+	--viewfinder-mode		Specify sensor mode, given as <width>:<height>:<bit-depth>:<packing>
+----
+
+This option is identical to the `--mode` option except that it applies only during the preview phase of stills capture (also used by the `libcamera-hello` application).
```
Comment (lurch) [inline documentation/asciidoc/accessories/camera/libcamera_options_common.adoc:182]: On this line you use "stills capture", on previous lines you used "still capture" - are they equivalent or would it be better to use just one?

File: documentation/asciidoc/accessories/camera/libcamera_options_common.adoc (hunk 3 above)
Comment (davidplowman) [reply documentation/asciidoc/accessories/camera/libcamera_options_common.adoc:37]: Long only. Will update that!

File: documentation/asciidoc/accessories/camera/libcamera_options_common.adoc (hunk 2 above)
Comment (davidplowman) [reply documentation/asciidoc/accessories/camera/libcamera_options_common.adoc:28]: Slightly inclined to leave it as `config.txt`, the option is `--config` after all. I'm hoping our users have learnt to cope with the proliferation of "configurations" everywhere...!

File: documentation/asciidoc/accessories/camera/libcamera_options_common.adoc (hunk 5 above)
Comment (davidplowman) [reply documentation/asciidoc/accessories/camera/libcamera_options_common.adoc:182]: Yes, tricky. I guess I prefer "stills capture" everywhere very slightly.

File: documentation/asciidoc/accessories/camera/libcamera_options_common.adoc (hunk 4 above)
Comment (davidplowman) [reply documentation/asciidoc/accessories/camera/libcamera_options_common.adoc:174]: Indeed. I think I'll go with "<n> bits" when "bits" is obviously a noun, but "<n>-bit" where it's combined as more of an adjective. So "12 bits per pixel" but "12-bit pixel". Does that seem plausible?

**Comment by davidplowman** [comment]
@lurch Thanks for the comments. I think that's mostly fixed up now...!

File: documentation/asciidoc/accessories/camera/libcamera_options_common.adoc (hunk 2 above)
Comment (lurch) [reply documentation/asciidoc/accessories/camera/libcamera_options_common.adoc:28]: ping @aallan - do you think this is something that users might get confused by?

Comment (aallan) [reply documentation/asciidoc/accessories/camera/libcamera_options_common.adoc:28]: I think that's okay, it's a fairly generic filename. I mean, that's not to say it isn't going to confuse *someone*. Everything always confuses *someone*. 🤷

**Comment by aallan** [comment]
Let me know when we're good to merge. Also whether it matters if I merge #2357 first, or this one first?

**Review by lurch — APPROVED** [review]

**Comment by aallan** [comment]
Merged #2357 first, we now have merge conflicts. Sorry folks. 🤦‍♂️

**Comment by naushir** [comment]
> Merged #2357 first, we now have merge conflicts. Sorry folks. man_facepalming

Looks like I won the race :-)

**Review by lurch — APPROVED** [review]

**Comment by aallan** [comment]
..and off we go!

**Comment by aallan** [comment]
It can sit on `develop` until the Bullseye update and then I'll push to prod.

**Comment by aallan** [comment]
@davidplowman Is this good to push to production now, or is there stuff that *really* needs to wait until the Bullseye update?  There is a couple of things on `develop` I'd like to push to `master` but I could cherry pick the commits if needed.

**Comment by davidplowman** [comment]
I'm happy to push the new stuff, I think the new release is in any case very imminent. Thanks!

**Pull request merged**